crates/cli/src/commands/install.rs

```diff
@@ -76,21 +80,26 @@ impl PackageManager {
     /// Jobs of the `install` command.
     pub async fn install(&self, args: &InstallCommandArgs) -> Result<(), PackageManagerError> {
         tracing::info!(target: "pacquet::install", "Start all");
+        let semaphore = Arc::new(Semaphore::new(16));
 
         self.package_json
             .dependencies(args.dependency_groups())
-            .map(|(name, version)| async move {
-                let dependency = find_package_version_from_registry(
-                    &self.tarball_cache,
-                    &self.config,
-                    &self.http_client,
-                    name,
-                    version,
-                    &self.config.modules_dir,
-                )
-                .await
-                .unwrap();
-                self.install_dependencies(&dependency).await;
+            .map(|(name, version)| {
+                let semaphore_clone = semaphore.clone();
```

Comment:
I advise `Arc::clone(&semaphore)` to avoid confusing it with regular expensive `.clone()`.